tjdbc/src/main/java/ru/tinkoff/zeppelin/tjdbc/JDBCInterpreter.java

```diff
@@ -0,0 +1,892 @@
+        logger.error("Failed to retrieve password from JCEKS \n" +
+            "For file: " + properties.getProperty(JDBC_JCEKS_FILE) +
+            "\nFor key: " + properties.getProperty(JDBC_JCEKS_CREDENTIAL_KEY), e);
+        throw e;
+      }
+    }
+    return null;
+  }
+
+  private String getResults(ResultSet resultSet, boolean isTableType, MutableBoolean isComplete)
+      throws SQLException {
+    ResultSetMetaData md = resultSet.getMetaData();
+    StringBuilder msg;
+    if (isTableType) {
+      msg = new StringBuilder(TABLE_MAGIC_TAG);
+    } else {
+      msg = new StringBuilder();
+    }
+
+    for (int i = 1; i < md.getColumnCount() + 1; i++) {
+      if (i > 1) {
+        msg.append(TAB);
+      }
+      if (StringUtils.isNotEmpty(md.getColumnLabel(i))) {
+        msg.append(replaceReservedChars(md.getColumnLabel(i)));
+      } else {
+        msg.append(replaceReservedChars(md.getColumnName(i)));
+      }
+    }
+    msg.append(NEWLINE);
+
+    int displayRowCount = 0;
+    while (resultSet.next()) {
+      if (displayRowCount >= getMaxResult()) {
+        isComplete.setValue(false);
+        break;
+      }
+      for (int i = 1; i < md.getColumnCount() + 1; i++) {
+        Object resultObject;
+        String resultValue;
+        resultObject = resultSet.getObject(i);
+        if (resultObject == null) {
+          resultValue = "null";
+        } else {
+          resultValue = resultSet.getString(i);
+        }
+        msg.append(replaceReservedChars(resultValue));
+        if (i != md.getColumnCount()) {
+          msg.append(TAB);
+        }
+      }
+      msg.append(NEWLINE);
+      displayRowCount++;
+    }
+    return msg.toString();
+  }
+
+  private boolean isDDLCommand(int updatedCount, int columnCount) throws SQLException {
+    return updatedCount < 0 && columnCount <= 0;
+  }
+
+  private InterpreterResult executeSql(String propertyKey, String sql,
+      InterpreterContext interpreterContext) {
+    Connection connection = null;
+    Statement statement;
+    ResultSet resultSet = null;
+    String paragraphId = interpreterContext.getParagraphId();
+    String user = interpreterContext.getAuthenticationInfo().getUser();
+
+    boolean splitQuery = false;
+    String splitQueryProperty = getProperty(String.format("%s.%s", propertyKey, SPLIT_QURIES_KEY));
+    if (StringUtils.isNotBlank(splitQueryProperty) && splitQueryProperty.equalsIgnoreCase("true")) {
+      splitQuery = true;
+    }
+
+    InterpreterResult interpreterResult = new InterpreterResult(InterpreterResult.Code.SUCCESS);
+    Optional<Connection> optionalConnection = establishConnection(interpreterContext, propertyKey);
+    if (!optionalConnection.isPresent()) {
+      try {
+        closeDBPool(user, propertyKey);
+      } catch (SQLException e1) {
+        logger.error("Cannot close DBPool for user, propertyKey: " + user + propertyKey, e1);
+      }
+      return new InterpreterResult(Code.ERROR, "Connection cannot be established");
```

Comment:
```suggestion
      return new InterpreterResult(Code.ERROR, "Database connection cannot be established");
```
Так как есть ещё connection между ядром и интерпретатором.
В оригинале там вообще выводят сообщение из `Exception`. Возможно, это даже хорошо. Надо подумать.

Reply:
OK, выбрал предложенный тобой вариант